### PR TITLE
[libmonodroid] Add a debug.mono.gdbwait property.

### PR DESCRIPTION
**Pull request by vargaz**
When this property is set, libmonodroid will wait in a loop for the native debugger to attach. The debugger can set a C variable to 0 to
break the wait after attaching. This is useful to support use cases where the debugger attaches to the process after it starts instead of
the debugger being started by libmonodroid.
The startup sequence should be:
* adb shell 'setprop debug.mono.gdbwait `date +%s`'
* adb shell 'am startup ...'
* <start debugger>
* <attach to process>
* p monodroid_gdb_wait=0

### CODE REVIEW

**Comment by jonpryor** [comment]
Should this be a new `debug.mono.gdbwait` property, or should we re-purpose the current `debug.mono.gdb` property? The latter isn't currently documented, so changing it's semantics shouldn't be problematic...

Actually, since you require that `debug.mono.gdbwait` be greater than 100000, I don't see why it should be a new property at all.

**Comment by vargaz** [comment]
debug.mono.gdb causes monodroid to start the gdbserver by itself, this property just causes it to wait for a debugger to attach. We can make the new property a subset of the old one, by using a special value like 'wait:<timestamp>'. Would that work ?

**Comment by vargaz** [comment]
Updated the PR.

**Comment by vargaz** [comment]
Does this look good ?

File: src/monodroid/jni/monodroid-glue.c [hunk 1]
```diff
@@ -2555,8 +2598,8 @@ static int
 should_attach_gdb (void)
 {
 	int i;
-	if (monodroid_get_namespaced_system_property (DEBUG_MONO_GDB_PROPERTY, NULL) > 0)
-		return 1;
+	if (!attach_gdb)
```
Comment (vargaz) [inline src/monodroid/jni/monodroid-glue.c]: parse_gdb_options () is called first.

**Comment by vargaz** [comment]
Updated according to the comments.

**Pull request merged**